asamisellah/Maintenance-Tracker
# #158113796 Setup Database and Refactor code

### What does this PR do?
Refactor the code to use blueprints and sets up connection of the app to the database 

### Description of Task to be completed?
- Install Postgres database(db)Create
- Connect db to api
- Create user table and requests table

** In refactor **
- Create a model file
- Create class User
- Create class Request
- Create user blueprint

### How should this be manually tested?
1. clone the [repo](https://github.com/asamisellah/Maintenance-Tracker)
2. open the folder in your text editor or IDE
3. `git checkout -b new-branch`
4. `git fetch ch-env-setup-157930685`

*Setup*
Make sure virtual environment is installed. Open terminal and run the following commands
1. `cd maintenance-tracker`
2.  `virtualenv venv` to create virtual environment
3.  `source venv/bin/activate` to run environment
4. `pip install -r requirements.txt`

*Database*
Prerequisites: Should have Postgres installed
Open terminal and run
``` $ sudo -u postgres psql
    $ CREATE DATABASE mtracker;
    $ \q   
    $ python schema.py
    $ python run.py
```
### Any background context you want to provide?
None

### What are the relevant pivotal tracker stories?
[#158141028](https://www.pivotaltracker.com/story/show/158141028/comments/190387102)
